### DISCUSSION TITLE
How can we see the reference file inside our repo after adding the file with git lfs?

### DISCUSSION THREAD

**Discussion by AizaTariq97** [git-lfs/git-lfs#5213] [Q&A]
I have pushed a file using git-lfs in github. After pushing it, I can only see .gitattribute file changed under my commit in github like this:

file_name filter=lfs diff=lfs merge=lfs -text 
There are couple of questions I want to ask:

1. Is .gitattribute file supposed to be changed only? Git lfs stores the reference of the file inside our repo. That reference file should be added in our repo right? Why can't I see that reference file under my commit in github?

2. Actual file is pushed in a different server. From where exactly can we view it?

3.Now when my branch gets merged with master branch, can other team members be able to access that file simply by adding git pull origin master? Obviously after they have properly configured git LFS.

4. git lfs ls-files is not showing any files after adding it. Why is this so?
 
The commands I followed to push files with git lfs are:

1.Install git LFS using this command:
sudo apt get install git-lfs

2. Initialize our repository with git-lfs using following command:
git lfs install

3. Track the files we want to add using this command:
git lfs track filename

4. Make sure .gitattributes is tracked using following command:
git add .gitattributes

5. Remaining steps are same:
git add file-name git commit -m "File added with git lfs" git push origin branch-name

**Comment by bk2204** [comment]
Hey,

You should also see the filename listed as a pointer file.  For example, the output of `git diff` on your commit might look something like this:

```
diff --git a/test1.bin b/test1.bin
new file mode 100644
index 0000000..1afab5f
--- /dev/null
+++ b/test1.bin
@@ -0,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:4355a46b19d348dc2f57c046f8ef63d4538ebb936000f3c9ee954a27460dd865
+size 2
```

Those three lines are a pointer file (obviously the object ID and the size will differ depending on the file).

If you're using GitHub or another hosting platform, the file is typically pushed to the same server.  If you've specifically set up a different server for LFS data, then it will be pushed there.

Typically, if you've run `git lfs install` on your local system properly, then when you check out a branch, either as part of a clone or otherwise, then the large files will be downloaded automatically and placed in the working tree.  Usually, Git will show the pointer file in `git diff` and such.  Checking the file out is the best way to view it; otherwise, you have to use [the API](https://github.com/git-lfs/git-lfs/tree/main/docs/api).

If you're not seeing any output in `git lfs ls-files` or a pointer file in `git show`, then it may be that there was a typo in tracking and the file wasn't tracked in Git LFS.  In that case, provide the output of `git show --name-only` and your `.gitattributes` file and we'll see if we can figure it out.

**Reply by AizaTariq97** [reply]
I have successfully added a file with git lfs, probably it was some kind of typo. Thanks a lot for explaining the concepts really well. Now it is updating two .gitattribute files, one with the location .gitattributes and other with the location My_folder/.gitattributes. My_folder is the name of folder where I have my file that is stored to track it with git lfs.

So my question is do we have to push both .gitattribute files in our repo? Both these files have identical changes.

**Reply by bk2204** [reply]
No, you don't need to have both.  You only need the `.gitattributes` file in the root of your repository, assuming they're identical and the pattern is some sort of wildcard (e.g., "*.pdf").

**Reply by AizaTariq97** [reply]
Yes they are identical, but the pattern is not wild card, it's just a single chrome driver file:
git lfs track chromedriver 

We have to run this command in root of our repository?

The pattern should be a wild card?

**Reply by bk2204** [reply]
A pattern like `chromedriver` or any path without a slash refers to that file pattern in the entire repository.  So `chromedriver` would be `chromedriver` in every directory.  If you add a path with a slash, then it's different and it's more restricted.  For example, `/foo/chromedriver` is the path `foo/chromedriver` directly under the root of the repository.

You can see information about the patterns available by running `git help gitattributes`, which refers you to `git help gitignore` (with a few changes).

Note also that Git is case-sensitive and that for gitattributes files that specifying a directory doesn't recursively match all paths inside that directory.
